src/dune_rules/lib.ml

```diff
@@ -1259,6 +1277,30 @@ end = struct
     ; runtime_deps : t list Resolve.Memo.t
     }
 
+  let resolve_ppx_runtime_deps =
+    let resolve_ppx_runtime_deps_impl (db, info, private_deps) =
+      resolve_simple_deps db (Lib_info.ppx_runtime_deps info) ~private_deps
+    in
+    let module Input = struct
+      type t = db * Path.t Lib_info.t * Private_deps.t
```

Comment:
Any change to the database is going to invalidate all the inputs with this database because of the physical equality, so how about we just move move this memo to the db itself:

```
; resolve_ppx_runtime_deps : ((Path.t Lib_info.t * Private_deps.t), _) Memo.t
```

This way the entire memo table can be gc'd if the db is gc'd.

Reply:
what's the `_` in your example?

Reply:
I'm a bit unsure how to store the memo for the ppx runtime libraries in the DB  (which DB?!). My understanding is that we discover these `ppx_runtime_deps` when instantiating the host library (because it's a PPX). so I'm a bit unsure how to resolve them during instantiation since we don't have the DB to resolve them.

I think I'm missing something in your suggestion.

Reply:
> what's the _ in your example?

It's the output of the memoized function. In this case, the list of resolved libraries.

>  My understanding is that we discover these ppx_runtime_deps when instantiating the host library (because it's a PPX). so I'm a bit unsure how to resolve them during instantiation since we don't have the DB to resolve them.

It's mostly a syntactic transformation. Everywhere where we have:

```
Memo.exec (db, info, private_deps)
```

We turn it into:

```
db.resolve_ppx_runtime_deps (info, private_deps
```

I imagine this should be possible

Reply:
PS: I realize I made a typo in my original comment that made it confusing. It should be `Memo.Table.t` rather than `Memo.t`

Reply:
aha, ok, I'll play with this.

Reply:
This is nicer, indeed. Moved the `resolve_ppx_runtime_deps` function to the `db` type.

Reply:
@rgrinberg I'm waiting for a final look after your suggestion before I merge this, to be sure.